### ISSUE TITLE
Automatically delete past events

### ISSUE THREAD

**Issue by nofishonfriday** [FossifyOrg/Calendar#295]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Calendar/issues) or [closed](https://github.com/FossifyOrg/Calendar/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Calendar/discussions) or [closed](https://github.com/FossifyOrg/Calendar/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Calendar/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

As suggested [here](https://github.com/FossifyOrg/Calendar/discussions/294), I'd like to propose an option to automatically delete past events/completed tasks.

### Why do you want this feature?

When events are past/tasks are completed, personally I have no need to store/see them anymore, so auto-deleting in this case would be convenient.
In fact I assumed this would be the case by default and was surprised when I recently by chance saw the long list of past events in the 'Simple event list' view.
Not a big thing, but they accumulate over time when not deleting manually occassionaly.

There's already an option to *dim* past events, so I guess *deleting* past events could ultilize some of the already present code.

**Comment by Aokromes** [comment]
imho such thing must wait few hours? day? before delete in case you forget it and you may reach event / taks even if late.

**Comment by nofishonfriday** [comment]
> imho such thing must wait few hours? day? before delete in case you forget it and you may reach event / taks even if late.

Yes, I meanwhile I think delete _on demand_ (an option/action which one can run from time) may be better than auto-delete.
With auto-delete it can easily be forgotten that it's enabled and then when wanting to keep a past event for whatever reason and it's gone that's no good.

**Comment by Aga-C** [comment]
@nofishonfriday Please adjust your original request to what exactly you'd like to have.

**Comment by nofishonfriday** [comment]
@Aga-C 
Thanks for the heads up, request updated.

(open to discuss implementation details with whomever wants to work on this of course. :))

**Comment by Aokromes** [comment]
> I think it would be nice either having an option/action 'delete all past events and completed tasks' the user can run from time to time or an auto-delete mechanism, though in this case concerns like mentioned from@Aokromes below should be taken into account.

maybe this can be a setting, from idk, 1h, 3h, 6h, 12h, 1d, 2d expired tasks?

**Comment by michalko58** [comment]
I would like this function too. BB calendar had this option for deleting calendar events older than 30 days. 

Here it could be adjustable for whatever period from 1 day to  1 year (?).

**Comment by sbstn87** [comment]
+1 for this feature

**Comment by xcpn** [comment]
I really want this implemented.

I have a calendar named Temps and I use it to create reminders for events that'll happen once and never again. Every time I need to go in the calendar and delete it afterwards.

Would really love an option while editing to delete after completion or after X days for example.

**Comment by Aokromes** [comment]
> I really want this implemented.
> 
> I have a calendar named Temps and I use it to create reminders for events that'll happen once and never again. Every time I need to go in the calendar and delete it afterwards.
> 
> Would really love an option while editing to delete after completion or after X days for example.

it's needed for example for you need to go to hospital day x, or school day y ora lot of single recurrence events.

**Comment by Anke** [comment]
I'd very much appreciate an option to delete all entries "older than".

**Comment by FakeTuba** [comment]
Request is 2 years old, what are the chances this will ever happen?
